src/app/routes/query.py

```diff
@@ -21,32 +20,69 @@ def route():
 
 
 def route_results():
-    if request.method == 'POST':
-        query_name = request.form.get('query_name') or uuid4()
-        query_text = request.form.get('query')
-        selected_auths = [auth for auth in request.form.getlist('auths') if auth]
+    # We need to handle our args in a few different ways, first and formost if they are passed as args in the url
+    arg_source = request.args.copy()
+
+    # Next, if we're given a form we should use that
+    if request.form:
+        arg_source.update(request.form)
+
+    # and finally if we're given json we should handle that
+    if request.is_json:
+        json_data = request.get_json()
+        if json_data:
+            arg_source.update(json_data)
+
+    # If we don't have any args at this point we should say there was no query to run
+    if not arg_source:
+        return render_template('query_result.html', no_query=True)
 
-        if not selected_auths:
-            flash('Please select at least one Auth option.', 'error')
-            return redirect(url_for('main.query'))
+    # Extract our arguments
+    query_name = arg_source.get('query_name') or uuid4()
+    query = arg_source.get('query')
+    data_type = arg_source.get('data_type')
+    decode_raw_data = bool(arg_source.get('decode_raw_data'))
```

Comment:
add a todo mentioning we need to add the other query parameters as options.

Reply:
Added a todo right after this.